bufferserver/src/main/java/com/datatorrent/bufferserver/internal/DataList.java

```diff
@@ -72,8 +72,9 @@
   private final AtomicInteger numberOfInMemBlockPermits;
   private MutableInt nextOffset = new MutableInt();
   private Future<?> future;
+  private final boolean backPressure;
 
-  public DataList(final String identifier, final int blockSize, final int numberOfCacheBlocks)
+  public DataList(final String identifier, final int blockSize, final int numberOfCacheBlocks, final boolean backPressure)
```

Comment:
I will go with backPressureEnable. I think spooling does have the effect of slowing things down but it comes from the memory limit requirement isn't it and will not result in back pressure if spooling is faster than downstream consumption.